packages/store/src/store.js

```diff
@@ -13,9 +13,12 @@ import { assert, details } from '@agoric/assert';
 function makeStore(keyName = 'key') {
   const store = new Map();
   const assertKeyDoesNotExist = key =>
-    assert(!store.has(key))([`${keyName} already registered`]);
+    assert(
+      !store.has(key),
+      details([`${keyName} already registered: `, ''], key),
+    );
   const assertKeyExists = key =>
-    assert(store.has(key))([`${keyName} not found: `, ''], key);
+    assert(store.has(key), details([`${keyName} not found: `, ''], key));
```

Comment:
this still seems way too clever. What would it be like using just normal assert usage?

Reply:
What do you mean "normal". It is using the calculated `keyName` both on the console and the thrown error. It is passing the `key` value only to the console and throwing only the typeof the `key` value. How would you write it?

Actually, I agree about it being too clever though. Here, it's encapsulated within the `assertKey*Exists` abstractions. But it appears again in `sameStructure`. I don't have a good idea.

Reply:
Among other things, this should not be constructing a template string dynamically in an assert. One of the typical goals of the assert logic (albeit minor for JS I know) is to minimize the allocations in the case where the assert passes. Thus, you create the details object but don't do any string mucking unless the assert is actually failed.

What's wrong with:
```assert(store.has(key), details`${keyName} ${key} not found`);```

Reply:
> What's wrong with...

The message in the thrown error would be
`(a string) (an object) not found`
when we'd like something more informative like
`asset (an object) not found`.

In any case, now fixed via `openDetail`. Done.